### PR TITLE
Adding failing tests for ISAM2 marginalization

### PR DESCRIPTION
**Pull request by gradyrw**
Adding unit tests in ISAM2 which trigger the failures described in https://github.com/borglab/gtsam/issues/1101

### CODE REVIEW

**Review by dellaert — APPROVED** [review]

**Comment by dellaert** [comment]
I can't merge as it is a draft PR :-) Remove draft status? I'll clean up after merging, as I try to understand the code.

**Comment by gradyrw** [comment]
Opened it for review! Just had to check that the results were the same w/ this code version (they are)

**Review by dellaert — APPROVED** [review]

**Pull request merged**
